collector/compression/zstd/zstd.go

```diff
@@ -0,0 +1,297 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package zstd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"runtime"
+	"sync"
+
+	zstdlib "github.com/klauspost/compress/zstd"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+)
+
+// NamePrefix is prefix, with N for compression level.
+const NamePrefix = "zstdarrow"
+
+// Level is an integer value mapping to compression level.
+// [0] implies disablement; not registered in grpc
+// [1,2] fastest i.e., "zstdarrow1", "zstdarrow2"
+// [3-5] default
+// [6-9] better
+// [10] best.
+type Level uint
+
+const (
+	DefaultLevel Level = 5
+	MinLevel     Level = 1
+	MaxLevel     Level = 10
+)
+
+type EncoderConfig struct {
+	// Level is meaningful in the range [0, 10].  No invalid
+	// values, they all map into 4 default configurations.
+	Level         Level  `mapstructure:"level"`
+	WindowSizeMiB uint32 `mapstructure:"window_size_mib"`
```

Comment:
Can you leave a comment about `WindowSizeMiB`, is this a memory limit?

Reply:
Done (all fields), mostly referring to the underlying zstd library method.